Datasets/bin/perl/isatab-search-replace.pl

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env perl
+#             -*- mode: cperl -*-
+#
+
+use strict;
+use warnings;
+use XML::LibXML;
+use Getopt::Long;
+use FindBin;
+use Tie::File;
+use File::Copy::Recursive qw(dircopy);
+
+#
+# usage:
+#
+#   ./isatab-search-replace.pl --studyName VBP0000123 --newVersion 2023-12-31 --search regexp --replace string
+#
+#
+# USE WITH CARE - it is very crude - use word delimiters in the search pattern to avoid unexpected replacements, e.g.
+#   --search '\bNCBITaxon_1234\b'
+# won't match NCBITaxon_12345.
+#
+# options:
+#
+#   --datasets ../../lib/xml/datasets/VectorBase.xml : path to datasets.xml (relative to script location)
+#   --manualDelivery /eupath/data/EuPathDB/manualDelivery/VectorBase/popbio : path to manualDelivery (this is the default)
+#   --dry-run : will copy the directory (but will not clobber) and make the search/replace edits
+#               but it won't update the datasets.xml file
+#
+# what does it do?
+#
+# 1. finds the <dataset> element in ./Datasets/lib/xml/datasets/VectorBase.xml with the requested studyName
+# 2. uses the current version name in that element to locate the (parent of the) ISA-Tab directory in manualDelivery,
+#    e.g. /eupath/data/EuPathDB/manualDelivery/VectorBase/popbio/<studyName>/2023-08-11
+# 3. copies that directory recursively to /eupath/data/EuPathDB/manualDelivery/VectorBase/popbio/<studyName>/<newVersion>
+# 4. applies a `s/search/replace/g` operation to all lines of all final/*.txt files in the new directory (no backup files made)
+# 5. updates the version string for the dataset in the XML representation
+# 6. outputs a new VectorBase.xml (not making a .old file - assuming file is under version control) (unless --dry-run)
```

Comment:
Shouldn't it also do 7? Also i dont understand the description in 3. Can you clarify?